services/src/kilda-core/kilda-persistence-api/src/main/java/org/openkilda/persistence/repositories/SwitchRepository.java

```diff
@@ -18,13 +18,16 @@
 import org.openkilda.model.Switch;
 import org.openkilda.model.SwitchId;
 
+import java.util.Collection;
 import java.util.Optional;
 
 public interface SwitchRepository extends Repository<Switch> {
     boolean exists(SwitchId switchId);
 
     Optional<Switch> findById(SwitchId switchId);
 
+    Collection<Switch> findByFlowId(String flowId);
```

Comment:
I'd recommend to rename the method to indicate what switches is fetched. E.g. findFlowEndpoints

Reply:
Fixed.